aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -272,61 +268,50 @@ mod loader {
+        /// Override the [`HttpClient`](aws_smithy_runtime_api::client::http::HttpClient) for this
+        /// [`ConfigLoader`]. The HTTP client will be used for both AWS services and credentials providers.
         ///
-        /// **Note**: In order to take advantage of late-configured timeout settings, you MUST use
-        /// [`HttpConnector::ConnectorFn`]
-        /// when configuring this connector.
+        /// If you wish to use a separate HTTP client for credentials providers when creating clients,
+        /// then override the HTTP client set with this function on the client-specific `Config`s.
         ///
-        /// If you wish to use a separate connector when creating clients, use the client-specific config.
         /// ## Examples
+        ///
         /// ```no_run
         /// # use aws_smithy_async::rt::sleep::SharedAsyncSleep;
-        /// use aws_smithy_client::http_connector::HttpConnector;
         /// #[cfg(feature = "client-hyper")]
         /// # async fn create_config() {
         /// use std::time::Duration;
-        /// use aws_smithy_client::{Client, hyper_ext};
-        /// use aws_smithy_client::erase::DynConnector;
-        /// use aws_smithy_client::http_connector::ConnectorSettings;
-        ///
-        /// let connector_fn = |settings:  &ConnectorSettings, sleep: Option<SharedAsyncSleep>| {
-        ///   let https_connector = hyper_rustls::HttpsConnectorBuilder::new()
-        ///       .with_webpki_roots()
-        ///       // NOTE: setting `https_only()` will not allow this connector to work with IMDS.
-        ///       .https_only()
-        ///       .enable_http1()
-        ///       .enable_http2()
-        ///       .build();
-        ///   let mut smithy_connector = hyper_ext::Adapter::builder()
-        ///       // Optionally set things like timeouts as well
-        ///       .connector_settings(settings.clone());
-        ///   smithy_connector.set_sleep_impl(sleep);
-        ///   Some(DynConnector::new(smithy_connector.build(https_connector)))
-        /// };
-        /// let connector = HttpConnector::ConnectorFn(std::sync::Arc::new(connector_fn));
+        /// use aws_smithy_runtime::client::http::hyper_014::HyperClientBuilder;
+        ///
+        /// let tls_connector = hyper_rustls::HttpsConnectorBuilder::new()
+        ///     .with_webpki_roots()
+        ///     // NOTE: setting `https_only()` will not allow this connector to work with IMDS.
+        ///     .https_only()
+        ///     .enable_http1()
+        ///     .enable_http2()
+        ///     .build();
+        ///
+        /// let hyper_client = HyperClientBuilder::new().build(tls_connector);
         /// let sdk_config = aws_config::from_env()
-        ///     .http_connector(connector)
+        ///     .http_client(hyper_client)
         ///     .load()
         ///     .await;
         /// # }
         /// ```
-        pub fn http_connector(mut self, http_connector: impl Into<HttpConnector>) -> Self {
```

Comment:
I played around with it, and I'm inclined to say this is a bad idea since it will lead to confusion around timeout configuration, which will be lost if this method is used, but not lost if the other method is used. I want there to be one right way to set this such that config works as expected.

Reply:
That makes more sense. I went ahead and added it back taking the HttpClient type with a deprecated message that points people in the right direction.